fix(ranges): Update thumb while animating container

dfelber opened the pull request:
To update the thumb while animating I am utilizing a `ResizeObserver` to trigger `translateThumbs`. This also should make the resize observation on the window superfluous.

closes: #79;

tajo commented:
Thanks for the PR. 

I think we should not add `resize-observer` as a dependency since it's quite a lot of code (related to this lib) and `ResizeObserver` is supported everywhere except IE.

Ideally, we could use `ResizeObserver` directly and have a fallback using the current solution when ResizeObserver is not supported. This way users can:

a. Ignore supporting IE and being ok with the current solution
b. Import the polyfill if they need to support IE

tajo commented on package.json:
```diff
@@ -59,6 +59,7 @@
     "jest-image-snapshot": "^2.7.0",
     "jest-puppeteer": "^4.3.0",
     "puppeteer": "^2.0.0",
+    "resize-observer": "^1.0.0",
```
If anything, it should be a normal dependency since you import it in the lib's code but I think we should not use it at all (my other comment).

dfelber replied:
The problem with not having the the `resize-observer` dependency is, that TypeScript does not know about ResizeObserver yet: https://github.com/microsoft/TypeScript/issues/37861
I am not familiar with TypeScript - maybe you know a better solution?

tajo replied:
We can just cast it to `any`. It's not an exposed through the lib's API anyway.

tajo commented on src/Range.tsx:
```diff
@@ -32,10 +33,11 @@ class Range extends React.Component<IProps> {
   schdOnMouseMove: (e: MouseEvent) => void;
   schdOnTouchMove: (e: TouchEvent) => void;
   schdOnEnd: (e: Event) => void;
-  schdOnWindowResize: () => void;
```
I like the `schdOnTrackResize` name but we should keep `schdOnWindowResize` as an alias for while so we don't have to introduce a breaking change. It seem pretty rare that you would resize the window without resizing the track at the same time so that should be fine.

dfelber commented:
> Thanks for the PR.
> 
> I think we should not add `resize-observer` as a dependency since it's quite a lot of code (related to this lib) and `ResizeObserver` is supported everywhere except IE.
> 
> Ideally, we could use `ResizeObserver` directly and have a fallback using the current solution when ResizeObserver is not supported. This way users can:
> 
> a. Ignore fully supporting IE and being ok with the current solution
> b. Import the polyfill if they _really_ need to support IE

Thanks for the feedback!
I updated the PR and removed the dependency to `resize-observer`.

tajo commented on src/Range.tsx:
```diff
@@ -60,7 +67,7 @@ class Range extends React.Component<IProps> {
   componentDidMount() {
     const { values, min, step } = this.props;
-    window.addEventListener('resize', this.schdOnWindowResize);
+    window.addEventListener('resize', this.schdOnResize);
```
Are we now triggering both `resize` event and `ResizeObserver`? This should not be added if ResizeObserver is supported?

dfelber replied:
I moved the resize event listeners to the `observe` and `unobserve` functions of the fallback object.

tajo commented:
Thanks! Published as `react-range@1.6.4`.
